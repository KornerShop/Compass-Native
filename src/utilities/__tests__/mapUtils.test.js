import {
  titleCase,
  findZipCode,
  fetchZipCode,
  fetchZipCodeCoords,
  fetchResults,
  fetchFoodBanks,
  fetchWICVendorDetails,
  fetchWICVendorsLocationPermission,
  fetchWICVendorsZipCode,
} from '../mapUtils';

test('titleCase', () => {
  const properlyCased = titleCase('ONE TWO THREE, 1ST, 2ND, 3RD');
  expect(properlyCased).toMatchSnapshot();
});

test('findZipCode', () => {
  const results = [
    {
      long_name: '2590',
      short_name: '2590',
      types: ['street_number'],
    },
    {
      long_name: 'Mark West Springs Road',
      short_name: 'Mark West Springs Rd',
      types: ['route'],
    },
    {
      long_name: 'Santa Rosa',
      short_name: 'Santa Rosa',
      types: ['locality', 'political'],
    },
    {
      long_name: 'Sonoma County',
      short_name: 'Sonoma County',
      types: ['administrative_area_level_2', 'political'],
    },
    {
      long_name: 'California',
      short_name: 'CA',
      types: ['administrative_area_level_1', 'political'],
    },
    {
      long_name: 'United States',
      short_name: 'US',
      types: ['country', 'political'],
    },
    {
      long_name: '95404',
      short_name: '95404',
      types: ['postal_code'],
    },
    {
      long_name: '9606',
      short_name: '9606',
      types: ['postal_code_suffix'],
    },
  ];
  const zip = findZipCode(results);
  expect(zip).toMatchSnapshot();
});

test('fetchZipCode', async () => {
  const location = {
    latitude: 38.5386881,
    longitude: -122.695547,
    latitudeDelta: 0.4,
    longitudeDelta: 0.4,
  };
  const zip = await fetchZipCode(location);
  expect(zip).toMatchSnapshot();
});

test('fetchZipCodeCoords', async () => {
  const coords = await fetchZipCodeCoords(95404);
  expect(coords).toMatchSnapshot();
});

describe('fetchResults', () => {
  it('returns an array of length 1 or more: SNAP', async () => {
    const results = await fetchResults(
      38.5386881,
      -122.695547,
      'calfresh',
    );
    expect(results.length).toBeGreaterThanOrEqual(1);
  });
  it('returns an array of length 1 or more: WIC', async () => {
    const results = await fetchResults(
      38.5386881,
      -122.695547,
      'wic',
    );
    expect(results.length).toBeGreaterThanOrEqual(1);
  });
  it('returns objects of proper shape: SNAP', async () => {
    const results = await fetchResults(
      38.5386881,
      -122.695547,
      'calfresh',
    );
    const firstResult = results[0];
    expect(firstResult).toHaveProperty('address');
    expect(firstResult).toHaveProperty('id');
    expect(firstResult).toHaveProperty('lat');
    expect(firstResult).toHaveProperty('lng');
    expect(firstResult).toHaveProperty('name');
    expect(firstResult).toHaveProperty('phone_intl');
    expect(firstResult).toHaveProperty('phone_local');
  });
  it('returns objects of proper shape: WIC', async () => {
    const results = await fetchResults(
      38.5386881,
      -122.695547,
      'wic',
    );
    const firstResult = results[0];
    expect(firstResult).toHaveProperty('address');
    expect(firstResult).toHaveProperty('id');
    expect(firstResult).toHaveProperty('lat');
    expect(firstResult).toHaveProperty('lng');
    expect(firstResult).toHaveProperty('name');
    expect(firstResult).toHaveProperty('phone_intl');
    expect(firstResult).toHaveProperty('phone_local');
  });
});

describe('fetchFoodBanks', () => {
  it('returns an array of length 1 or more', async () => {
    const results = await fetchFoodBanks(38.5386881, -122.695547);
    expect(results.length).toBeGreaterThanOrEqual(1);
  });
  it('returns objects of proper shape', async () => {
    const results = await fetchFoodBanks(38.5386881, -122.695547);
    const firstResult = results[0];
    expect(firstResult).toHaveProperty('address');
    expect(firstResult).toHaveProperty('id');
    expect(firstResult).toHaveProperty('lat');
    expect(firstResult).toHaveProperty('lng');
    expect(firstResult).toHaveProperty('name');
    expect(firstResult).toHaveProperty('phone_intl');
    expect(firstResult).toHaveProperty('phone_local');
  });
});

describe('fetchVendorDetails', () => {
  it('returns an array of length 1 or more', async () => {
    const results = await fetchWICVendorDetails(95404);
    expect(results.length).toBeGreaterThanOrEqual(1);
  });
  it('returns objects of proper shape', async () => {
    const results = await fetchWICVendorDetails(95404);
    const firstResult = results[0];
    expect(firstResult).toHaveProperty('address');
    expect(firstResult).toHaveProperty('id');
    expect(firstResult).toHaveProperty('lat');
    expect(firstResult).toHaveProperty('lng');
    expect(firstResult).toHaveProperty('name');
  });
});

describe('fetchWICVendorsLocationPermission', () => {
  it('returns an array of length 1 or more', async () => {
    const results = await fetchWICVendorsLocationPermission(
      38.5386881,
      -122.695547,
    );
    expect(results.length).toBeGreaterThanOrEqual(1);
  });
  it('returns objects of proper shape', async () => {
    const results = await fetchWICVendorsLocationPermission(
      38.5386881,
      -122.695547,
    );
    const firstResult = results[0];
    expect(firstResult).toHaveProperty('address');
    expect(firstResult).toHaveProperty('id');
    expect(firstResult).toHaveProperty('lat');
    expect(firstResult).toHaveProperty('lng');
    expect(firstResult).toHaveProperty('name');
  });
});

describe('fetchWICVendorsZipCode', () => {
  it('returns an array of length 1 or more', async () => {
    const results = await fetchWICVendorsZipCode(95404);
    expect(results.length).toBeGreaterThanOrEqual(1);
  });
  it('returns objects of proper shape', async () => {
    const results = await fetchWICVendorsZipCode(95404);
    const firstResult = results[0];
    expect(firstResult).toHaveProperty('address');
    expect(firstResult).toHaveProperty('id');
    expect(firstResult).toHaveProperty('lat');
    expect(firstResult).toHaveProperty('lng');
    expect(firstResult).toHaveProperty('name');
  });
});
