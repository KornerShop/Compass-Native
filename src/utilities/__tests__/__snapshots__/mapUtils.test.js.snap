// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchResults 1`] = `
Array [
  Object {
    "address": "2550 Paulin Dr # 1, Santa Rosa, CA 95403, USA",
    "id": "ChIJWcnjX3c4hIAReEaEIu-h5Dg",
    "lat": 38.464769,
    "lng": -122.721573,
    "name": "Sonoma County Human Services Department",
    "phone_intl": "+1 877-699-6868",
    "phone_local": "(877) 699-6868",
  },
  Object {
    "address": "520 Mendocino Ave, Santa Rosa, CA 95402, USA",
    "id": "ChIJ2S8f1PhHhIARUVCsEnyrOEw",
    "lat": 38.443262,
    "lng": -122.715587,
    "name": "Sonoma County Economic Assistance 2",
    "phone_intl": "+1 877-699-6868",
    "phone_local": "(877) 699-6868",
  },
]
`;

exports[`fetchZipCode 1`] = `"95404"`;

exports[`fetchZipCodeCoords 1`] = `
Object {
  "lat": 38.5386881,
  "lng": -122.695547,
}
`;

exports[`findZipCode 1`] = `"95404"`;
